config/configmapprovider/retrieved.go

```diff
@@ -21,38 +21,42 @@ import (
 	"go.opentelemetry.io/collector/config"
 )
 
-// Retrieved holds the result of a call to the Retrieve method of a Provider object.
-// This interface cannot be directly implemented. Implementations must use the NewRetrieved helper.
-type Retrieved interface {
-	// Get returns the config Map. Should never be called after Close.
-	// Should never be called concurrently with itself or Close.
-	Get(ctx context.Context) (*config.Map, error)
-
-	// Close signals that the configuration for which it was used to retrieve values is
-	// no longer in use and should close and release any watchers that it may have created.
-	//
-	// Should block until all resources are closed, and guarantee that `onChange` is not
-	// going to be called after it returns except when `ctx` is cancelled.
-	//
-	// Should never be called concurrently with itself or Get.
-	Close(ctx context.Context) error
+// Get returns the config Map. Should never be called after Close.
+// Should never be called concurrently with itself or Close.
+//
+// Deprecated: Not needed, will be removed soon. You have access to the struct members.
+func (r Retrieved) Get(ctx context.Context) (*config.Map, error) {
+	return r.Map, nil
+}
 
-	// privateRetrieved is an unexported func to disallow direct implementation.
-	privateRetrieved()
+// Close signals that the configuration for which it was used to retrieve values is
+// no longer in use and should close and release any watchers that it may have created.
+//
+// Should block until all resources are closed, and guarantee that `onChange` is not
+// going to be called after it returns except when `ctx` is cancelled.
+//
+// Should never be called concurrently with itself or Get.
+//
+// Deprecated: Not needed, will be removed soon. You have access to the struct members.
```

Comment:
Don't you think this one can still be useful?

Reply:
Why? May be... but I want to limit the public api "when in doubt, leave it out" :)

Reply:
I mean r.Close() seems cleaner than r.CloseFunc() with a condition :) 

If we are considering adding it later maybe makes sense to keep it now

Reply:
That is the whole idea, I am in doubt :) sooo.. I prefer to leave it out.

Reply:
Are you 100% convince that we need that "helper" func?

Reply:
I tried one more option, but still not convinced that this helper is really needed.